src/galaxy/SectorGenerator.cpp

```diff
@@ -13,10 +13,164 @@
+		for (int i = 0; i < len; i++) {
+			name += sys_names[rng.Int32(0, SYS_NAME_FRAGS - 1)];
+		}
+
+		name[0] = toupper(name[0]);
+	}
+} // namespace HybridNames
+
+namespace Doomdark {
+	static const char *Prefixes[] = {
+		"img", "dol", "lor", "ush", "mor", "tal", "car", "ulf", "as", "tor", "ob", "f", "gl",
+		"s", "th", "gan", "mal", "im", "var", "hag", "zar", "anv", "ber", "kah", "ash"
+	};
+	static const unsigned int PREFIX_FRAGS = ((unsigned int)(sizeof(Prefixes) / sizeof(char *)));
+
+	static const char *Midwords[] = {
+		"ar", "or", "ir", "en", "orth", "angr", "igr", "ash", "el", "in", "ul", "atr", "orm", "udr", "is", "ildr"
+	};
+	static const unsigned int MIDWORD_FRAGS = ((unsigned int)(sizeof(Midwords) / sizeof(char *)));
+
+	static const char *Suffixes[] = {
+		"orn", "il", "iel", "im", "uk", "ium", "ia", "eon", "ay", "ak", "arg", "and", "ane", "esh", "ad", "un"
+	};
+	static const unsigned int SUFFIX_FRAGS = ((unsigned int)(sizeof(Suffixes) / sizeof(char *)));
+
+	void GetName(std::string &name, Random &rng)
+	{
+		// Doodarken a name
+		name += Prefixes[rng.Int32(0, PREFIX_FRAGS - 1)];
+		name += Midwords[rng.Int32(0, MIDWORD_FRAGS - 1)];
+		name += Suffixes[rng.Int32(0, SUFFIX_FRAGS - 1)];
+
+		name[0] = toupper(name[0]);
+	}
+} // namespace Doomdark
+
+namespace Katakana {
+	// clang-format off
+	static const char *StartFragments[] = {
+		"kyo","gyo","shu","sho","chu","cho","hyu","myo",
+		"ryu","chi","tsu","shi","ka","ki","ku","ke",
+		"ko","ga","gi","gu","ge","go","sa","su",
+		"se","so","za","zu","ze","ta","te","to",
+		"da","na","ni","nu","ne","no","ha","hi",
+		"fu","he","ho","ba","bi","bu","be","ma",
+		"mi","mu","me","mo","ya","yu","yo","ri",
+		"ru","wa","jo","a","i","u","e","o",
+		
+	};
+	static const char *MiddleFragments[] = {
+		"sshi","ppo","tto","mbo","kka","kyu","sho","chu",
+		"chi","tsu","shi","ka","ki","ku","ke","ko",
+		"ga","gi","gu","ge","go","sa","su","se",
+		"so","za","ji","zu","ze","ta","te","to",
+		"da","de","do","na","ni","nu","ne","no",
+		"ha","hi","fu","ho","ba","bi","bu","be",
+		"bo","ma","mi","mu","me","mo","ya","yo",
+		"ra","ri","ru","re","ro","wa","ju","jo",
+		"a","i","u","e","o","n",
+	};
+	static const char *EndFragments[] = {
+		"ttsu","ppu","ssa","tto","tte","noh","mba","kko",
+		"kyo","shu","chu","nyu","nyo","ryu","chi","tsu",
+		"shi","ka","ki","ku","ke","ko","ga","gi",
+		"gu","go","sa","su","se","so","za","ji",
+		"zu","ze","zo","ta","te","to","da","de",
+		"do","na","ni","ne","no","ha","hi","fu",
+		"he","ho","ba","bi","bu","be","bo","ma",
+		"mi","mu","me","mo","ya","yo","ra","ri",
+		"ru","re","ro","wa","ja","jo","i","e",
+		"o","n",
+	};
+	// clang-format on
+
+	static const unsigned int NUM_START_FRAGS = ((unsigned int)(sizeof(StartFragments) / sizeof(char *)));
+	static const unsigned int NUM_MIDDLE_FRAGS = ((unsigned int)(sizeof(MiddleFragments) / sizeof(char *)));
+	static const unsigned int NUM_END_FRAGS = ((unsigned int)(sizeof(EndFragments) / sizeof(char *)));
```

Comment:
Unless I'm horribly mistaken, we have a `COUNTOF` macro to use in the case where we need to get the length of an array at compile time. I'd recommend converting the manual array length calculations to use that macro for simplicity.

Reply:
Yes, I think I'm the one who added it 🤦 I'll change that.